data/stories/chitchat.yml

```diff
@@ -732,9 +734,13 @@ stories:
     entities:
     - product: nlu
   - action: utter_ask_for_nlu_specifics
-  - intent: nlu_info
-    entities:
-    - nlu_part: intent classification
+  - or:
+    - intent: nlu_info
+      entities:
+      - nlu_part: intent classification
+    - intent: explain
+      entities:
+      - nlu_part: intent classification
```

Comment:
Do the NLU examples need to be updated for the explain intent to extract the `intent classification` and `entity recognition` entities?

Reply:
Good question. However sensible it is, I might actually disagree: Looking at the NLU inbox, I can't find real examples classified as `explain` that would have those entities. On the other hand, there's a lot of real examples classified as `nlu_info` which do sound close to `intent:explain` too, e.g.:
```
No explain entity recognition
can you explain duckling to me
explain intents
explain how to create entity file
```

Assuming we really want to keep classifying these as `nlu_info`, I wouldn't move any of them in the training data for `explain`. Then, I'd treat the relevant training story as a safety net for cases where NLU predicts `explain` -- so that Core still does the right thing then.

After all, my adding of `explain` in places where `nlu_info` is in stories was inspired by several existing training stories where there was an `or` between `nlu_info` and `explain`. In other words, this adding wasn't (in some cases) inspired by any real conversations, but rather by this "safety net" pattern.

Reply:
I think it'd make sense to move `nlu_info` into `explain` since there'd be quite a bit of ambiguity otherwise. The `nlu_part` entity would have to be renamed to something more general but IMO it'd be a much cleaner way of handling all these cases.

Reply:
In the context of this PR, I think either the `explain` intent should be re-factored to handle the scope of `nlu_info` or the safety net pattern should be left out for now to avoid confusion from a best practices perspective.

Reply:
Thanks @b-quachtran, makes sense. I guess the issue is that we think/care about two different things.
- I think primarily about the new Sara test set, which isn't supposed to be perfect, will inevitably become out-dated soon, and should be used with a particular "archived" snapshot of `rasa-demo` (for model regression purposes).
- You care about `rasa-demo`, which is an evolving thing.

Personally, for the purposes of the new Sara test set, I would just stop here, perhaps reverting the "safety net" changes. But for `rasa-demo`, I totally agree with merging `nlu_info` and `explain` (and altering the `nlu_part` entity). I guess what I'll do for now is leaving only those training story changes in this PR, which were inspired by real conversations (i.e. no "safety net" stuff). And then merging `nlu_info` and `explain` can be another issue, which won't be related to the new Sara test set at all. Sounds good?

Reply:
Alright, I've reverted the changes that weren't backed by real stories we have at hand today.